context/app/static/js/components/workspaces/WorkspaceToastSuccessMessage.tsx

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import Button from '@mui/material/Button';
+
+function WorkspaceToastSuccessMessage(id: number) {
+  return (
+    <Stack sx={{ maxWidth: '22rem' }}>
+      <Typography>
+        Workspace successfully launched in a new tab. If the tab didn&apos;t open, please check your pop-up blocker
+        settings and relaunch your workspace.
+      </Typography>
+      <Stack direction="row" justifyContent="flex-end" marginTop={1}>
+        <Button
+          href={`/workspaces/${id}`}
+          variant="contained"
+          sx={{
+            backgroundColor: 'transparent',
+            boxShadow: 'none',
+            ':hover': { backgroundColor: 'transparent', boxShadow: 'none' },
+          }}
+        >
+          <Typography variant="button">View Workspace Detail Page</Typography>
+        </Button>
```

Comment:
Would it be possible to extend the `message` object that's sent to the snackbar by the various `toast` functions to accept an `action` and pass that in to the `Alert` in `StyledSnackbar`? This would allow us to use the API provided by MUI/avoid needing to add these extra styles: https://mui.com/material-ui/react-alert/#actions

Reply:
It looks like it is possible, but some restyling of the main `toast` component would be needed in order to match the designs. Right now, the `action` slot is being used for a close button, and any additional `action`s passed in will override that, so there would need to be logic to handle both cases. Additionally, the extra styling for this button would still need to be there to match designs.

If this seems like an option that we could reuse frequently, I think it's doable, but otherwise I'd stay with the current React Node `message`.

<img width="1262" alt="Screenshot 2024-08-21 at 3 56 30 PM" src="https://github.com/user-attachments/assets/749f71ac-f3cf-40a7-873e-d1c50432a6a8">